src/services/RPCSelector.ts

```diff
@@ -0,0 +1,69 @@
+import { JsonRpcProvider, Block } from '@ethersproject/providers';
+import { isTimestampMoreRecentThan } from '../utils/helpers';
+
+interface ProviderComparand {
+  blockNumber: number;
+  url: string;
+}
+
+class RPCSelector {
+  readonly urls: string[];
+  readonly preferredProviderUrl: string;
```

Comment:
let's drop preferredProviderUrl. urls are ordered by the priority.

Reply:
The `preferredProviderUrl` is there precisely because it's the prioritized one. I've wanted to make it explicit on the code.